[5.x] Expose field conditions from GraphQL API

duncanmcclean opened the pull request:
This pull request exposes field conditions from the GraphQL forms resource. 

Closes #11600.

duncanmcclean commented:
Test added ✅

jasonvarga reviewed (changes requested):
You can edit `tests/Feature/GraphQL/FormTest.php@it_queries_the_fields` instead.

Please make `if` and `unless` fall back to `null` instead of empty arrays.

The `validate` values are inconsistent with what the REST API returns. I think maybe we just skip adding validate to the fields array items. It wasn't requested in the linked issue.

If anything, we should add a `validate` array to the top level form that has all the fields. The way each field can potentially give validation rules for other fields doesn't make sense in a loop, it's used by Statamic to compile the complete rule list.

duncanmcclean commented:
@jasonvarga I've removed the validation rules (it looks like they're available at the top-level anyways), and I've updated the tests. Ready for re-review.
